### PR TITLE
feat(ci): Add CI tasks to push/pull from transifex

### PR DESCRIPTION
**Pull request by billyvg**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,13 +69,13 @@ compile-locale:
 	cd src/sentry && sentry django compilemessages
 
 install-transifex:
-	$(PIP) install transifex-client
+	[ -f ./tx ] || curl -o- https://raw.githubusercontent.com/transifex/cli/master/install.sh | bash
```
Comment (billyvg) [inline Makefile:72]: via https://developers.transifex.com/docs/cli#how-to-install-the-client

File: .github/workflows/pull-transifex.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: transifex
+on:
+  workflow_dispatch:
```
Comment (billyvg) [inline .github/workflows/pull-transifex.yml:3]: The pull workflow is not in a state where we can reliably run it on a schedule. There are some issues with our translations (e.g. use of literal `%`) that will cause this workflow to error.

**Review by billyvg — COMMENTED** [review]
I included the changes when running `make push-transifex` locally.

File: Makefile (hunk 1 above)
Comment (asottile-sentry) [inline Makefile:72]: curl bash makes me nervous -- I would prefer we control the script which does this download if possible -- looks like it mostly pulls from here https://github.com/transifex/cli/releases

we should download from there and checksum -- we can get away with only supporting (linux, mac), (x86_64, arm64) probably -- I'd recommend a small python similar to setup-volta: https://github.com/getsentry/action-setup-volta/blob/main/bin/setup-volta (can probably put this in `tools/`)

**Review by asottile-sentry — COMMENTED** [review]
otherwise looks good!

**Pull request closed without merging**